app/assets/v2/js/amounts.js

```diff
@@ -0,0 +1,48 @@
+var estimate= function(amount, conv_rate) {
+    var estimateAmount = amount / conv_rate;
+    if (estimateAmount) {
+        estimateAmount = Math.round(estimateAmount * 100) / 100;
+        if (estimateAmount > 1000000) {
+            estimateAmount = Math.round(estimateAmount / 100000) / 10 + "m"
+        } else if (estimateAmount > 1000) {
+            estimateAmount = Math.round(estimateAmount / 100) / 10 + "k"
+        }
+        return ('Approx: ' + estimateAmount + ' USD');
+    } else {
+        return null;
+    }
+};
+
+var getUSDEstimate = function (amount, denomination) {
+    var conv_rate;
+    var usd_amount;
+    var eth_amount;
+    try {
+        amount = parseFloat(amount);
+    } catch (e) {
+        return null;
+    }
+
+    if (document.conversion_rates && document.conversion_rates[denomination]) {
+        conv_rate = document.conversion_rates[denomination]
+        return estimate(amount, conv_rate)
```

Comment:
itd be great if, on error or missing data, the UI was smart enough to update to say "uknown amount" instead of just keeping the last value.